### PR TITLE
Preview ChangeFeed: Fixes headers on failure path.

### PR DESCRIPTION
**Pull request by bchong95**
# Pull Request Template

## Preview ChangeFeed: Fixes headers on failure path.

Some headers were missing. In the future we need a generic ResponseMessage to CosmosException utility method.

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Pagination/NetworkAttachedDocumentContainer.cs [hunk 1]
```diff
@@ -361,11 +361,17 @@ internal sealed class NetworkAttachedDocumentContainer : IMonadicDocumentContain
             else
             {
                 CosmosException cosmosException = new CosmosException(
-                    responseMessage.ErrorMessage,
                     statusCode: responseMessage.StatusCode,
-                    (int)responseMessage.Headers.SubStatusCode,
-                    responseMessage.Headers.ActivityId,
-                    responseMessage.Headers.RequestCharge);
+                    message: responseMessage.ErrorMessage,
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Pagination/NetworkAttachedDocumentContainer.cs:365]: https://github.com/Azure/azure-cosmos-dotnet-v3/blob/122203cebf5d9127f31a70b99657fe16b6de1fd9/Microsoft.Azure.Cosmos/src/Resource/CosmosExceptions/CosmosExceptionFactory.cs#L68

File: Microsoft.Azure.Cosmos/src/Pagination/NetworkAttachedDocumentContainer.cs [hunk 2]
```diff
@@ -219,20 +220,7 @@ internal sealed class NetworkAttachedDocumentContainer : IMonadicDocumentContain
...
-                    responseMessage.Headers.ActivityId,
-                    responseMessage.Headers.RequestCharge,
-                    responseMessage.Headers.RetryAfter,
-                    responseMessage.Headers,
-                    error: null,
-                    innerException: null,
-                    trace: trace);
-                cosmosException.Headers.ContinuationToken = responseMessage.Headers.ContinuationToken;
-
+                CosmosException cosmosException = CosmosExceptionFactory.Create(responseMessage);
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Pagination/NetworkAttachedDocumentContainer.cs:223]: What about the trace? Would we dropping/missing the traces with this change?

**Comment by kirankumarkolli** [comment]
Please add UT coerage.

File: Microsoft.Azure.Cosmos/src/Pagination/NetworkAttachedDocumentContainer.cs (hunk 2 above)
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Pagination/NetworkAttachedDocumentContainer.cs:223]: Can we add some unit tests covering for this scenario to assert the current or new behavior regarding the headers so we avoid any future change accidentally regressing the fact that headers should be present?

**Review by vivekr20 — APPROVED** [review]

File: Microsoft.Azure.Cosmos/src/Pagination/NetworkAttachedDocumentContainer.cs (hunk 2 above)
Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Pagination/NetworkAttachedDocumentContainer.cs:223]: It's up to the CosmosExceptionFactory method to copy over the trace by reference.

Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Pagination/NetworkAttachedDocumentContainer.cs:223]: Compute said they would add the tests.

Comment (j82w) [reply Microsoft.Azure.Cosmos/src/Pagination/NetworkAttachedDocumentContainer.cs:223]: The tests need to be done in the github repo. Having tests on the internal repository causes painful delays on the compute sync, and allows regression to be shipped in the public SDK.

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/Pagination/NetworkAttachedDocumentContainer.cs:223]: +1. Our first customers are public users, we should avoid accidentally shipping a regression in the first place. Relying on a future verification on a sync cadence that is independent of our release schedule can hurt our public users.

Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Pagination/NetworkAttachedDocumentContainer.cs:223]: This has been discussed offline.

**Review by sboshra — APPROVED** [review]

**Pull request merged**